### PR TITLE
Refactors manage.py

### PR DESCRIPTION
**Pull request by Ryxias**
to: @ryandeivert @chunyong-lin @blakemotl 
cc: @airbnb/streamalert-maintainers
related to: #677

## Background

While building out lookup tables, I noticed that `manage.py` is complex and fragmented in its functionality.

This PR refactors manage.py to make it easier to add/manage commands. It brings the subparsers closer to the handlers, which reduces the number of places a developer has to look to add/remove/modify commands.

## Changes

* New `CliCommand` abstract class that all commands inherit from.

### CODE REVIEW

File: stream_alert_cli/apps/handler.py [hunk 1]
```diff
@@ -13,97 +13,232 @@
...
 """
+import string
+
 from stream_alert.apps import StreamAlertApp
 from stream_alert.shared.logger import get_logger
 from stream_alert_cli.apps.helpers import save_app_auth_info
+from stream_alert_cli.utils import (
+    CliCommand, set_parser_epilog, generate_subparser,
+    add_schedule_expression_arg, add_timeout_arg, add_memory_arg,
+    CLUSTERS)
```
Comment (ryandeivert) [inline stream_alert_cli/apps/handler.py]: hanging `)` should be on new line

File: stream_alert_cli/utils.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+import textwrap
+from stream_alert.apps.config import AWS_RATE_RE, AWS_RATE_HELPER
+
+CLUSTERS = [
+    os.path.splitext(cluster)[0] for _, _, files in os.walk('../conf/clusters')
+    for cluster in files
+]
+
+
+class CliCommand:
```
Comment (ryandeivert) [inline stream_alert_cli/utils.py]: maybe a nit .. but I think `CLICommand` is more pythonic (?)

File: stream_alert_cli/runner.py [hunk 3]
```diff
@@ -13,27 +13,27 @@
...
-    terraform_clean_handler,
-    terraform_destroy_handler,
-    terraform_init,
-    terraform_list_targets
-)
-from stream_alert_cli.test.handler import test_handler
-from stream_alert_cli.threat_intel_downloader.handler import threat_intel_downloader_handler
+from stream_alert_cli.manage_lambda.deploy import DeployCommand
+from stream_alert_cli.manage_lambda.rollback import RollbackCommand
+from stream_alert_cli.metrics_alarms.handler import CreateMetricAlarmCommand, \
```
Comment (ryandeivert) [inline stream_alert_cli/runner.py]: please wrap in parenthesis and omit the `\`:

```python
from stream_alert_cli.metrics_alarms.handler import (
    CreateMetricAlarmCommand,
    CreateClusterMetricAlarmCommand,
    CustomMetricsCommand
)
```

File: stream_alert_cli/runner.py [hunk 4]
```diff
@@ -13,27 +13,27 @@
...
+    CreateClusterMetricAlarmCommand, CustomMetricsCommand
+from stream_alert_cli.outputs.handler import OutputCommand
+from stream_alert_cli.rule_table import RuleStagingCommand
+from stream_alert_cli.status.handler import StatusCommand
+from stream_alert_cli.terraform.generate import TerraformGenerateCommand
+from stream_alert_cli.terraform.handlers import TerraformCleanCommand, TerraformBuildCommand, \
```
Comment (ryandeivert) [inline stream_alert_cli/runner.py]: same as above

Comment (ryandeivert) [reply stream_alert_cli/runner.py]: for reasoning behind this nit: http://google.github.io/styleguide/pyguide.html#32-line-length

specifically: `Do not use backslash line continuation except for with statements requiring three or more context managers.`

File: stream_alert_cli/apps/handler.py (hunk 1 above)
Comment (ryandeivert) [reply stream_alert_cli/apps/handler.py]: it's also good practice to keep imports in alphabetical order

File: stream_alert_cli/athena/handler.py [hunk 5]
```diff
@@ -20,6 +20,8 @@
 from stream_alert.shared.logger import get_logger
 from stream_alert_cli.athena import helpers
 from stream_alert_cli.helpers import continue_prompt, record_to_schema
+from stream_alert_cli.utils import CliCommand, generate_subparser, set_parser_epilog, \
```
Comment (ryandeivert) [inline stream_alert_cli/athena/handler.py]: same newline continuation comment

File: stream_alert_cli/apps/handler.py (hunk 1 above)
Comment (Ryxias) [reply stream_alert_cli/apps/handler.py]: Sorry, I miss these a lot because my IDE auto-generates these imports and also auto-hides them

Comment (ryandeivert) [reply stream_alert_cli/apps/handler.py]: bad IDE. BAD!

**Pull request merged**